### ISSUE TITLE
Zoomed images are translucent on OSX in ff 38

### ISSUE THREAD

**Issue by GaryKing** [dadler/thumbnail-zoom#215]
This is a particularly bad bug (is it a bug?). Zoomed images are now transcluent, so if there is a dark image behind the zoomed image, for instance, then the zoomed image itself will also be darker. I noticed this starting two days or so ago. I'm using Firefox 38.0.1 on OS X 10.10.3.

**Comment by dadler** [comment]
Yeah, it's nothing I changed; apparently Firefox 38 decided my pop-ups should use the OSX transparency feature.  

Workaround: turn on "System Preferences > Accessibility > Display > Reduce transparency".

**Comment by dadler** [comment]
Below is an example.  With "Reduce transparency" off (default), the pop-up in Firefox 38 is darker and bluer than the embedded image.  With "Reduce transparency" on, the pop-up is true black and white, matching the embedded image.

![example](http://upload.wikimedia.org/wikipedia/commons/a/ad/Afterimage.svg)

**Comment by GaryKing** [comment]
Bummer, I would have liked to keep Reduce Transparency enabled, but if it
negatively affects TZP, then the latter has priority, since it's actually
useful, while the former is just nice to have.

**Comment by dadler** [comment]
Searching for first bad firefox version via:
mozregression -g 2015-01-02 -b 2015-05-15 -p "/Users/dadler/Library/Application Support/Firefox/Profiles/p0976zws.default"

[2015-02-05, 2015-02-06] (1 days) (~0 steps left)
23:07.15 LOG: MainThread Bisector INFO Got as far as we can go bisecting nightlies...
23:07.15 LOG: MainThread Bisector INFO Last good revision: 34a66aaaca81
23:07.15 LOG: MainThread Bisector INFO First bad revision: 7c5f187b65bf
23:07.15 LOG: MainThread Bisector INFO Pushlog:
https://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=34a66aaaca81&tochange=7c5f187b65bf

**Comment by dadler** [comment]
99:46.45 LOG: MainThread Bisector INFO Last good revision: b3fa9d6b1310
99:46.45 LOG: MainThread Bisector INFO First bad revision: b46cb1ef16f8
99:46.45 LOG: MainThread Bisector INFO Pushlog:
https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?fromchange=b3fa9d6b1310&tochange=b46cb1ef16f8

Changes pushed after changeset b3fa9d6b1310, up to and including changeset b46cb1ef16f8
User
Push date [To Local]	Changeset	Patch author — Commit message
mstange@themasta.com
Wed Feb 04 22:25:37 2015 +0000	b46cb1ef16f8	Markus Stange — Bug 1045213 - Make menu separators look correct by having them use foreground vibrancy. r=smichaud
	caac1a006b9d	Markus Stange — Bug 1045213 - Add a MakeRegionsNonOverlapping function for less repetitive code. r=botond
	c07194e580ce	Markus Stange — Bug 1045213 - Get smooth rounded corners for vibrant context menus by using _cornerMask. r=smichaud
	fa950339690a	Markus Stange — Bug 1045213 - Make the highlighted menu item vibrant. r=smichaud
	4e3291b79e57	Markus Stange — Bug 1045213 - Add nsITheme::ThemeGeometryTypeForWidget. r=roc
	984a21c9423c	Markus Stange — Bug 1045213 - Use vibrancy effect for context menus. r=smichaud
	657fc2d2da4a	Markus Stange — Bug 1045213 - Make ns(Int)Region methods chainable. r=jrmuizel

So the bug is caused by one of those changes, which are for this issue:
[Bug 1045213](https://bugzilla.mozilla.org/show_bug.cgi?id=1045213) - [10.10] Contextmenus shadow and background color are incorrect

**Comment by GaryKing** [comment]
Okay great. Ideally, it'd be nice to have "Reduce transparency" disabled and have TZP working fine as it was before, but for now I've just enabled the setting so that TZP works.

**Comment by dadler** [comment]
This is fixed (I believe) in [TZP 3.5beta1](https://addons.mozilla.org/en-US/firefox/addon/thumbnail-zoom-plus/versions/), which I just released.

I discovered the fix by bisecting Firefox versions to find the patch set which broke TZP, and then reading the diffs of each commit in the patch set.  A comment in the diffs mentioned CSS -moz-appearance, which I then googled to see its options.  Tried it and explicitly setting it to "none" worked!
